lib/ui/window/pointer_data_packet_converter.cc

```diff
@@ -186,7 +186,16 @@ void PointerDataPacketConverter::ConvertPointerData(
         auto iter = states_.find(pointer_data.device);
         FML_DCHECK(iter != states_.end());
         PointerState state = iter->second;
-        FML_DCHECK(state.is_down);
+        if (!state.is_down) {
```

Comment:
The synthetic down be added in the embedding side. This issues a click event to the framework if engine receives an up event without down. This can be problematic because different platforms may have different expectations. For example, what if iOS sends an up event in a different situation where it expect the up to be dropped instead of issuing a click?